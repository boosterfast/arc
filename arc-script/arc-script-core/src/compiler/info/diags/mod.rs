pub mod repr;
pub mod to_codespan;
pub mod sink;
pub(crate) mod utils;

pub use repr::*;
pub use sink::*;
