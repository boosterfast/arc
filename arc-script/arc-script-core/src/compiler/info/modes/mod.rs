pub mod repr;
/// Module for representing sink and source information.
pub mod connector;

pub use repr::*;
