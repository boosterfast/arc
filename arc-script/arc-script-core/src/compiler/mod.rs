/// Module for representing Arc Queries.
pub(crate) mod query;
/// Module for representing the Abstract Syntax Tree.
///
/// NB: This module is public so that external libraries can construct ASTs
/// without having to generate source code.
pub mod ast;
/// Module for representing Dataflow graphs. Exports data types for staging
/// `HIR` functions.
pub mod dfg;
/// Module for representing Higher Order Intermediate Representation code.
pub(crate) mod hir;
/// Module for representing side information.
pub mod info;
/// Module for representing Multi-Level Intermediate Representation code.
pub(crate) mod mlir;
/// Module for representing Arcon code.
pub(crate) mod arcon;

// Module for incremental compilation.
// pub(crate) mod database;
/// Module which assembles compilation-pipeline.
pub mod pipeline;

pub(crate) mod pretty;

pub use pipeline::compile;
