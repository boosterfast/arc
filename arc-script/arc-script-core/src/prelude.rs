//! Exports for users who use the compiler as a library.

pub use crate::compiler;
pub use crate::compiler::ast;
pub use crate::compiler::info::diags;
pub use crate::compiler::info::logger;
pub use crate::compiler::info::modes;

pub use anyhow::Result;
pub use half::bf16;
pub use half::f16;
