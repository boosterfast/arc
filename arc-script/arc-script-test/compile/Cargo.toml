[package]
name        = "arc-script-test-compile"
version     = "0.0.0"
authors     = ["Klas Segeljakt <klasseg@kth.se>"]
edition     = "2018"
license     = "MIT"
description = "Tests to see if arc-scripts compile"

[dependencies]
trybuild        = { version = "1.0.39" }
arc-script-core = { version = "=0.0.0", path = "../../arc-script-core" }
insta           = { version = "1.6.0", features = ["glob"] }

[dev-dependencies]
arcon           = { git = "https://github.com/cda-group/arcon" }
arc-script      = { version = "=0.0.0", path = "../../" }
prost           = { version = "0.7.0" }
