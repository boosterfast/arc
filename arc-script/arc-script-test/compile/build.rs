fn main() {
    arc_script_build::Builder::default()
        .no_exclude(true)
        .source_dirs(["src/tests/expect_pass", "src/tests/expect_mlir_fail_todo"])
        .build();
}
