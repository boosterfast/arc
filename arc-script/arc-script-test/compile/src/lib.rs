#![allow(unused)]

#[cfg(test)]
mod insta;

// #[cfg(test)]
// mod trybuild;

// #[cfg(test)]
// mod compiletest;

mod tests {
    mod expect_pass {
        mod basic_pipe;
        mod binops;
        mod enum_pattern;
        mod enum_pattern_nested;
        mod enums;
        mod fib;
        mod fun;
        mod if_let;
        mod ifs;
        mod lambda;
        mod literals;
        mod nested_if;
        mod option;
        mod path;
        mod pattern;
        mod r#if;
        mod structs;
    }
    mod expect_mlir_fail_todo {
        mod extern_fun;
        mod pipe;
        mod stateful;
        mod task_filter;
        mod task_identity_untagged;
        mod task_map;
        mod task_with_funs;
    }
}
