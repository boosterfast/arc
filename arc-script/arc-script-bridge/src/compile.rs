use arc_script_core::compiler::compile;
use arc_script_core::compiler::info::diags::sink::Buffer;
use arc_script_core::prelude::modes::{Input, Mode, Output};

use proc_macro as pm;
use quote::quote;

use std::fs;
use std::path::PathBuf;

/// arc_script::compile!("script.arc")
///    .stage_fun1(1, 2, 3)
///    .stage_fun2("foo", "bar")
///    .finalize()
pub(crate) fn expand(input: pm::TokenStream) -> pm::TokenStream {
    todo!()
}
