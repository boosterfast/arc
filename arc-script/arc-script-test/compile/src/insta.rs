#![cfg_attr(not(test), allow(dead_code, unused_imports))]

use arc_script_core::prelude::compiler::compile;
use arc_script_core::prelude::diags::Buffer;
use arc_script_core::prelude::modes::{Input, Mode, Output};

use std::env;

fn snapshot(output: Output, paths: &str) {
    env::set_var("INSTA_OUTPUT", "summary");
    env::set_var("INSTA_FORCE_PASS", "1");
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_snapshot_suffix();
    let mut sink = Buffer::no_color();
    settings.bind(|| {
        insta::glob!(paths, |path| {
            println!("Testing {}", path.display());
            let mode = Mode {
                input: Input::File(Some(path.into())),
                output,
                ..Default::default()
            };
            compile(mode, &mut sink).unwrap();
            let s = std::str::from_utf8(sink.as_slice()).unwrap();
            insta::assert_snapshot!(s);
            sink.clear();
        })
    });
}

#[test]
fn test_ast() {
    snapshot(Output::AST, "tests/expect_mlir_fail_todo/*.arc");
    snapshot(Output::AST, "tests/expect_pass/*.arc");
    snapshot(Output::AST, "tests/expect_fail/*.arc");
    snapshot(Output::AST, "tests/expect_fail_todo/*.arc");
}

#[test]
fn test_hir() {
    snapshot(Output::HIR, "tests/expect_mlir_fail_todo/*.arc");
    snapshot(Output::HIR, "tests/expect_pass/*.arc");
    snapshot(Output::HIR, "tests/expect_fail/*.arc");
    snapshot(Output::HIR, "tests/expect_fail_todo/*.arc");
}

#[test]
fn test_dfg() {
    snapshot(Output::DFG, "tests/expect_mlir_fail_todo/*.arc");
    snapshot(Output::DFG, "tests/expect_pass/*.arc");
    snapshot(Output::DFG, "tests/expect_fail/*.arc");
    snapshot(Output::DFG, "tests/expect_fail_todo/*.arc");
}

#[test]
fn test_rust() {
    snapshot(Output::Rust, "tests/expect_mlir_fail_todo/*.arc");
    snapshot(Output::Rust, "tests/expect_pass/*.arc");
    snapshot(Output::Rust, "tests/expect_fail/*.arc");
    snapshot(Output::Rust, "tests/expect_fail_todo/*.arc");
}

#[test]
fn test_mlir() {
    snapshot(Output::MLIR, "tests/expect_pass/*.arc");
    snapshot(Output::MLIR, "tests/expect_fail/*.arc");
    snapshot(Output::MLIR, "tests/expect_fail_todo/*.arc");
}
