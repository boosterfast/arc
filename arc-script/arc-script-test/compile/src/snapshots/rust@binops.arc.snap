---
source: arc-script-test/compile/src/insta.rs
expression: s
input_file: arc-script-test/compile/src/tests/expect_pass/binops.arc

---
#[allow(non_snake_case)]
#[allow(unused_must_use)]
#[allow(dead_code)]
#[allow(unused_variables)]
#[allow(unused_imports)]
#[allow(unused_braces)]
pub mod defs {
    use super::*;
    pub use arc_script::arcorn;
    pub use arc_script::arcorn::state::{ArcMapOps, ArcRefOps, ArcSetOps, ArcVecOps};
    pub use arcon::prelude::*;
    pub fn main() -> () {
        let y_1_0 = true;
        let y_1_1 = 127i8;
        let y_1_2 = 32767i16;
        let y_1_3 = 2147483647i32;
        let y_1_4 = 9223372036854775807i64;
        let y_1_5 = 255u8;
        let y_1_6 = 65535u16;
        let y_1_7 = 4294967295u32;
        let y_1_8 = 18446744073709551615u64;
        let y_1_9 = 340282350000000000000000000000000000000f32;
        let y_1_10 = 179769313486231570000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000f64 ;
        let y_1_11 = 126i8;
        let y_1_12 = 32766i16;
        let y_1_13 = 2147483646i32;
        let y_1_14 = 9223372036854775806i64;
        let y_1_15 = 254u8;
        let y_1_16 = 65534u16;
        let y_1_17 = 4294967294u32;
        let y_1_18 = 18446744073709551614u64;
        let y_1_19 = 34028235000000000000000000000000000000f32;
        let y_1_20 = 179769313486231550000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000f64 ;
        let y_1_21 = y_1_1 + y_1_1;
        let y_1_22 = y_1_2 + y_1_2;
        let y_1_23 = y_1_3 + y_1_3;
        let y_1_24 = y_1_4 + y_1_4;
        let y_1_25 = y_1_5 + y_1_5;
        let y_1_26 = y_1_6 + y_1_6;
        let y_1_27 = y_1_7 + y_1_7;
        let y_1_28 = y_1_8 + y_1_8;
        let y_1_29 = y_1_9 + y_1_9;
        let y_1_30 = y_1_10 + y_1_10;
        let y_1_31 = y_1_1 - y_1_1;
        let y_1_32 = y_1_2 - y_1_2;
        let y_1_33 = y_1_3 - y_1_3;
        let y_1_34 = y_1_4 - y_1_4;
        let y_1_35 = y_1_5 - y_1_5;
        let y_1_36 = y_1_6 - y_1_6;
        let y_1_37 = y_1_7 - y_1_7;
        let y_1_38 = y_1_8 - y_1_8;
        let y_1_39 = y_1_9 - y_1_9;
        let y_1_40 = y_1_10 - y_1_10;
        let y_1_41 = y_1_1 * y_1_1;
        let y_1_42 = y_1_2 * y_1_2;
        let y_1_43 = y_1_3 * y_1_3;
        let y_1_44 = y_1_4 * y_1_4;
        let y_1_45 = y_1_5 * y_1_5;
        let y_1_46 = y_1_6 * y_1_6;
        let y_1_47 = y_1_7 * y_1_7;
        let y_1_48 = y_1_8 * y_1_8;
        let y_1_49 = y_1_9 * y_1_9;
        let y_1_50 = y_1_10 * y_1_10;
        let y_1_51 = y_1_1 / y_1_1;
        let y_1_52 = y_1_2 / y_1_2;
        let y_1_53 = y_1_3 / y_1_3;
        let y_1_54 = y_1_4 / y_1_4;
        let y_1_55 = y_1_5 / y_1_5;
        let y_1_56 = y_1_6 / y_1_6;
        let y_1_57 = y_1_7 / y_1_7;
        let y_1_58 = y_1_8 / y_1_8;
        let y_1_59 = y_1_9 / y_1_9;
        let y_1_60 = y_1_10 / y_1_10;
        let y_1_61 = y_1_1 % y_1_1;
        let y_1_62 = y_1_2 % y_1_2;
        let y_1_63 = y_1_3 % y_1_3;
        let y_1_64 = y_1_4 % y_1_4;
        let y_1_65 = y_1_5 % y_1_5;
        let y_1_66 = y_1_6 % y_1_6;
        let y_1_67 = y_1_7 % y_1_7;
        let y_1_68 = y_1_8 % y_1_8;
        let y_1_69 = y_1_9 % y_1_9;
        let y_1_70 = y_1_10 % y_1_10;
        let y_1_71 = y_1_1 <= y_1_1;
        let y_1_72 = y_1_2 <= y_1_2;
        let y_1_73 = y_1_3 <= y_1_3;
        let y_1_74 = y_1_4 <= y_1_4;
        let y_1_75 = y_1_5 <= y_1_5;
        let y_1_76 = y_1_6 <= y_1_6;
        let y_1_77 = y_1_7 <= y_1_7;
        let y_1_78 = y_1_8 <= y_1_8;
        let y_1_79 = y_1_9 <= y_1_9;
        let y_1_80 = y_1_10 <= y_1_10;
        let y_1_81 = y_1_1 < y_1_1;
        let y_1_82 = y_1_2 < y_1_2;
        let y_1_83 = y_1_3 < y_1_3;
        let y_1_84 = y_1_4 < y_1_4;
        let y_1_85 = y_1_5 < y_1_5;
        let y_1_86 = y_1_6 < y_1_6;
        let y_1_87 = y_1_7 < y_1_7;
        let y_1_88 = y_1_8 < y_1_8;
        let y_1_89 = y_1_9 < y_1_9;
        let y_1_90 = y_1_10 < y_1_10;
        let y_1_91 = y_1_1 > y_1_1;
        let y_1_92 = y_1_2 > y_1_2;
        let y_1_93 = y_1_3 > y_1_3;
        let y_1_94 = y_1_4 > y_1_4;
        let y_1_95 = y_1_5 > y_1_5;
        let y_1_96 = y_1_6 > y_1_6;
        let y_1_97 = y_1_7 > y_1_7;
        let y_1_98 = y_1_8 > y_1_8;
        let y_1_99 = y_1_9 > y_1_9;
        let y_1_100 = y_1_10 > y_1_10;
        let y_1_101 = y_1_1 >= y_1_1;
        let y_1_102 = y_1_2 >= y_1_2;
        let y_1_103 = y_1_3 >= y_1_3;
        let y_1_104 = y_1_4 >= y_1_4;
        let y_1_105 = y_1_5 >= y_1_5;
        let y_1_106 = y_1_6 >= y_1_6;
        let y_1_107 = y_1_7 >= y_1_7;
        let y_1_108 = y_1_8 >= y_1_8;
        let y_1_109 = y_1_9 >= y_1_9;
        let y_1_110 = y_1_10 >= y_1_10;
        let y_1_111 = y_1_0 == y_1_0;
        let y_1_112 = y_1_1 == y_1_1;
        let y_1_113 = y_1_2 == y_1_2;
        let y_1_114 = y_1_3 == y_1_3;
        let y_1_115 = y_1_4 == y_1_4;
        let y_1_116 = y_1_5 == y_1_5;
        let y_1_117 = y_1_6 == y_1_6;
        let y_1_118 = y_1_7 == y_1_7;
        let y_1_119 = y_1_8 == y_1_8;
        let y_1_120 = y_1_9 == y_1_9;
        let y_1_121 = y_1_10 == y_1_10;
        let y_1_122 = y_1_0 != y_1_0;
        let y_1_123 = y_1_1 != y_1_1;
        let y_1_124 = y_1_2 != y_1_2;
        let y_1_125 = y_1_3 != y_1_3;
        let y_1_126 = y_1_4 != y_1_4;
        let y_1_127 = y_1_5 != y_1_5;
        let y_1_128 = y_1_6 != y_1_6;
        let y_1_129 = y_1_7 != y_1_7;
        let y_1_130 = y_1_8 != y_1_8;
        let y_1_131 = y_1_9 != y_1_9;
        let y_1_132 = y_1_10 != y_1_10;
        let y_1_133 = y_1_0 && y_1_0;
        let y_1_134 = y_1_0 || y_1_0;
        let y_1_135 = y_1_0 ^ y_1_0;
        let y_1_136 = y_1_1 & y_1_1;
        let y_1_137 = y_1_2 & y_1_2;
        let y_1_138 = y_1_3 & y_1_3;
        let y_1_139 = y_1_4 & y_1_4;
        let y_1_140 = y_1_5 & y_1_5;
        let y_1_141 = y_1_6 & y_1_6;
        let y_1_142 = y_1_7 & y_1_7;
        let y_1_143 = y_1_8 & y_1_8;
        let y_1_144 = y_1_1 | y_1_1;
        let y_1_145 = y_1_2 | y_1_2;
        let y_1_146 = y_1_3 | y_1_3;
        let y_1_147 = y_1_4 | y_1_4;
        let y_1_148 = y_1_5 | y_1_5;
        let y_1_149 = y_1_6 | y_1_6;
        let y_1_150 = y_1_7 | y_1_7;
        let y_1_151 = y_1_8 | y_1_8;
        let y_1_152 = y_1_1 ^ y_1_1;
        let y_1_153 = y_1_2 ^ y_1_2;
        let y_1_154 = y_1_3 ^ y_1_3;
        let y_1_155 = y_1_4 ^ y_1_4;
        let y_1_156 = y_1_5 ^ y_1_5;
        let y_1_157 = y_1_6 ^ y_1_6;
        let y_1_158 = y_1_7 ^ y_1_7;
        let y_1_159 = y_1_8 ^ y_1_8;
        let y_1_160 = y_1_9.powf(y_1_19);
        let y_1_161 = y_1_10.powf(y_1_20);
        let y_1_162 = ();
        y_1_162
    }
}
pub use defs::*;


